Explorer/Assets/DCL/AvatarRendering/Wearables/Systems/ResolveWearableByPointerSystem.cs

```diff
@@ -183,32 +188,61 @@ private void FinalizeWearableDTO([Data] bool defaultWearablesResolved, in Entity
             {
                 if (!promiseResult.Succeeded)
                 {
+                    //No wearable representation is going to be possible 
                     foreach (string pointerID in promise.LoadingIntention.Pointers)
-                    {
-                        wearableCatalog.TryGetWearable(pointerID, out IWearable component);
-                        SetDefaultWearables(defaultWearablesResolved, component, in bodyShape);
-                        component.IsLoading = false;
-                    }
+                        ReportAndFinalizeWithError(pointerID);
                 }
                 else
                 {
+                    var failedDTOList = WearableComponentsUtils.POINTERS_POOL.Get();
+                    failedDTOList.AddRange(promise.LoadingIntention.Pointers);
+                    
                     foreach (WearableDTO assetEntity in promiseResult.Asset.Value)
                     {
-                        if (!wearableCatalog.TryGetWearable(assetEntity.metadata.id, out IWearable component))
+                        bool isWearableInCatalog = wearableCatalog.TryGetWearable(assetEntity.metadata.id, out var component);
+
+                        try
+                        {
+                            Assert.IsTrue(isWearableInCatalog); //A wearable that has a DTO request should already have an empty representation in the catalog at this point
```

Comment:
- `Assert` is conditional `[Conditional("UNITY_ASSERTIONS")]` so it won't be present in prod builds
- Let's not abuse exceptions catching system, you don't need to throw to print an error (moreover you ignore the exception anyway)